fix: Fix CSS inconsistency in Contributor Highlights for "Home" and "Following" tabs for `/feed` route

5hraddha opened the pull request:
## Description
This PR fixes CSS inconsistency in Contributor Highlight Cards for "Home" and "Following" tabs for `/feed` route.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
Fixes #1778

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/27571141/e9e82d49-71df-4efd-83e3-672e6e3eb16f

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?



## [optional] What gif best describes this PR or how it makes you feel?



<!-- note: PRs with deleted sections will be marked invalid -->

<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
  
  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.
  
  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the Open Sauced Contributing Guide: https://github.com/open-sauced/.github/blob/main/CONTRIBUTING.md.
  - 📖 Read the Open Sauced Code of Conduct: https://github.com/open-sauced/.github/blob/main/CODE_OF_CONDUCT.md.
  - 👷‍♀️ Create small PRs. In most cases, this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

nickytonline commented:
Thanks for the PR @5hraddha! Mind adding a before and after screenshot or video in the description?

5hraddha commented:
> Mobile & Desktop Screenshots/Recordings



> Thanks for the PR @5hraddha! Mind adding a before and after screenshot or video in the description?

Sure @nickytonline! I've added a video recording in _Mobile & Desktop Screenshots/Recordings_ section after the change, but I'll do for the before the change too. Thanks!

bdougie commented:
> I've added a video recording in _Mobile & Desktop Screenshots/Recordings_ section after the change, but I'll do for the before the change too. Thanks!

I embedded them to save a click

bdougie reviewed (approved):
Looks good.

5hraddha commented:
> > I've added a video recording in _Mobile & Desktop Screenshots/Recordings_ section after the change, but I'll do for the before the change too. Thanks!
> 
> I embedded them to save a click

Thanks @bdougie!

jpmcb reviewed (approved):
Thanks for diving into this!! 🍕 🚀

jpmcb reviewed (changes requested):
Looks like, on diving deeper, this is still a problem with responsive, smaller screen sizes:


https://github.com/open-sauced/app/assets/23109390/8593db55-8b92-4213-ba1d-ae5366031b0c

Notice when I make the screen abit smaller and the feed collapses in abit, the problem persists. On the other responsive screens (like for the smaller mobile view), it seems fine. Just the first resizing still seems broken.

Maybe there's an additional class we're missing?

5hraddha commented:
> (apologies for the re-review) Looks like, on diving deeper, this is still a problem with responsive, smaller screen sizes:
> 
>  resized-home-following.mov 
> Notice when I make the screen abit smaller and the feed collapses in abit, the problem persists. On the other responsive screens (like for the smaller mobile view), it seems fine. Just the first resizing still seems broken.
> 
> Maybe there's an additional class we're missing?

Thanks for noticing that @jpmcb. I've fixed the glitch. Here is the recording for the reference. Please, let me know if that is suffice. Thanks!


https://github.com/open-sauced/app/assets/27571141/668c3d8a-4141-4b85-9673-a8eb7dc35d60

OgDev-01 reviewed (approved):
LGTM

jpmcb reviewed (changes requested):
https://github.com/open-sauced/app/assets/23109390/d8a287c1-ed37-4e0a-b1ec-dca372e4a5c1

It looks like there's still one view that doesn't have the same styling.

Notice the 3rd resizing in my screen capture still has a problem.

5hraddha commented:
> jpmcb



> 2023-10-03.15-36-58.mp4 
> It looks like there's still one view that doesn't have the same styling.
> 
> Notice the 3rd resizing in my screen capture still has a problem.

Hey @jpmcb, Could you please help me with the screen size where the CSS breaks?

jpmcb commented:
Sure! If I go less than 1280px wide, it drops to a smaller size without the user profile column on the left:

<img width="1168" alt="Screenshot 2023-10-03 at 4 14 57 PM" src="https://github.com/open-sauced/app/assets/23109390/3bfd390a-9aad-4b35-a440-cf796a0b484c">

NsdHSO reviewed (approved):
Good to ship.

5hraddha commented:
> Sure! If I go less than 1280px wide, it drops to a smaller size without the user profile column on the left:
> 
> <img alt="Screenshot 2023-10-03 at 4 14 57 PM" width="1168" src="https://user-images.githubusercontent.com/23109390/272410630-3bfd390a-9aad-4b35-a440-cf796a0b484c.png">

Am sorry that this is prolonging this far. But, it's bit weird @jpmcb.  I could see different behaviours in different browsers. On Chrome, it's all good at screen width smaller than 1280px. Notice the sizes of the card on inspection. They are same. Attaching a recording for reference:

https://github.com/open-sauced/app/assets/27571141/82921918-b32f-4918-b833-c5e6ea7955c3

On Firefox, I do see some negligible glitch. On inspection alone, I saw that there is a minute difference in the sizes of the card. 

https://github.com/open-sauced/app/assets/27571141/247aa57a-03d2-4519-8b38-585731c64556

Which browser are you checking on?

5hraddha commented:
Am sorry that this is prolonging this far. But, it's bit weird @jpmcb.  I could see different behaviours in different browsers. On Chrome, it's all good at screen width smaller than 1280px. Notice the sizes of the card on inspection. They are same. Attaching a recording for reference:

https://github.com/open-sauced/app/assets/27571141/82921918-b32f-4918-b833-c5e6ea7955c3

On Firefox, I do see some negligible glitch. On inspection alone, I saw that there is a minute difference in the sizes of the card. 

https://github.com/open-sauced/app/assets/27571141/247aa57a-03d2-4519-8b38-585731c64556

Which browser are you checking on?

nickytonline commented:
> Which browser are you checking on?

@jpmcb is using Firefox, so we should ensure it works well there as well.

5hraddha commented:
@nickytonline @jpmcb I figured out that there was a bug in `tailwind.config.js` while adding the additional smallest breakpoint - `xs`. As per the [Tailwind documentation](https://tailwindcss.com/docs/screens#adding-smaller-breakpoints), if we want to add an additional small breakpoint, we can’t use `extend` because the small breakpoint would be added to the end of the breakpoint list, and breakpoints need to be sorted from smallest to largest in order to work as expected with a `min-width` breakpoint system. So, I have updated the config as suggested in the documentation. 

Once that was done, my breakpoints were working as they should without the surprising behaviour. Then, I have added extra needed Tailwind classes to the component, to make sure they are rendered the same way in both the browsers.

Attaching the screen recording for reference:

**Chrome**

https://github.com/open-sauced/app/assets/27571141/4a3bbf0b-6b31-48ba-a4c2-1e5e259ee528

**Firefox**


https://github.com/open-sauced/app/assets/27571141/e8db9758-56c5-4bff-9750-586b4a6f1c23


I believe that should resolve the issue. Please, have a check and let me know if you find any other unusual CSS behaviour in this regard. Thank you!

jpmcb commented:
Nice investigation on this!!!

I still see the same problem on the smaller view now only on Firefox:

#### Home feed (which seems to be slightly wider)

![Screenshot 2023-10-10 at 1 53 25 PM](https://github.com/open-sauced/app/assets/23109390/cd4a16a7-b5dc-4284-a9c4-00044c18d720)

#### Following feed (which seems to be slightly narrower)

![Screenshot 2023-10-10 at 1 53 10 PM](https://github.com/open-sauced/app/assets/23109390/bd957b9f-03dd-4acf-b638-97e1554a586b)

we can merge this in and try to roll forward a fix for the firefox edge case if people prefer

5hraddha commented:
Thanks for your patience, @jpmcb. Appreciate your keen eye on UI, too. 

I just feel that there is still something missing that I'm not able to find at the moment. The UI should not be this quirky in Firefox. I'll keep looking into this and will raise an issue if I happen to find the problem.

Thanks again!
